src/System.Net.NetworkInformation/src/System/Net/NetworkInformation/LinuxIPGlobalProperties.cs

```diff
@@ -0,0 +1,392 @@
+            parser.MoveNextOrFail(); // skip Index
+
+            string localAddressAndPort = parser.MoveAndExtractNext(); // local_address
+            IPAddress localAddress;
+            int localPort;
+            ParseAddressAndPort(localAddressAndPort, out localAddress, out localPort);
+
+            string remoteAddressAndPort = parser.MoveAndExtractNext(); // rem_address
+            IPAddress remoteAddress;
+            int remotePort;
+            ParseAddressAndPort(remoteAddressAndPort, out remoteAddress, out remotePort);
+
+            string socketStateHex = parser.MoveAndExtractNext();
+            int state;
+            if (!int.TryParse(socketStateHex, NumberStyles.HexNumber, CultureInfo.InvariantCulture, out state))
+            {
+                throw new InvalidOperationException("Invalid state value: " + socketStateHex);
+            }
+            TcpState tcpState = MapTcpState((Interop.LinuxTcpState)state);
+
+            IPEndPoint localEndPoint = new IPEndPoint(localAddress, localPort);
+            IPEndPoint remoteEndPoint = new IPEndPoint(remoteAddress, remotePort);
+
+            return new SimpleTcpConnectionInformation(localEndPoint, remoteEndPoint, tcpState);
+        }
+
+        // Maps from Linux TCP states (include/net/tcp_states.h) to .NET TcpStates
+        // TODO: Move this to the native shim.
+        private static TcpState MapTcpState(Interop.LinuxTcpState state)
+        {
+            switch (state)
+            {
+                case Interop.LinuxTcpState.TCP_ESTABLISHED:
+                    return TcpState.Established;
+                case Interop.LinuxTcpState.TCP_SYN_SENT:
+                    return TcpState.SynSent;
+                case Interop.LinuxTcpState.TCP_SYN_RECV:
+                    return TcpState.SynReceived;
+                case Interop.LinuxTcpState.TCP_FIN_WAIT1:
+                    return TcpState.FinWait1;
+                case Interop.LinuxTcpState.TCP_FIN_WAIT2:
+                    return TcpState.FinWait2;
+                case Interop.LinuxTcpState.TCP_TIME_WAIT:
+                    return TcpState.TimeWait;
+                case Interop.LinuxTcpState.TCP_CLOSE:
+                    return TcpState.Closing;
+                case Interop.LinuxTcpState.TCP_CLOSE_WAIT:
+                    return TcpState.CloseWait;
+                case Interop.LinuxTcpState.TCP_LAST_ACK:
+                    return TcpState.LastAck;
+                case Interop.LinuxTcpState.TCP_LISTEN:
+                    return TcpState.Listen;
+                case Interop.LinuxTcpState.TCP_CLOSING:
+                    return TcpState.Closing;
+                case Interop.LinuxTcpState.TCP_NEW_SYN_RECV:
+                    return TcpState.Unknown;
+                case Interop.LinuxTcpState.TCP_MAX_STATES:
+                    return TcpState.Unknown;
+                default:
+                    throw new InvalidOperationException("Invalid LinuxTcpState: " + state);
```

Comment:
Shouldn't we just default to `TcpState.Unknown` instead? These can change in time although the implementation should still work.
We could `Debug.Assert` instead to trigger a potential platform update.